### ISSUE TITLE
Main thread stall in whitelist mode.

### ISSUE THREAD

**Issue by clienthax** [SpongePowered/SpongeForge#3111]
**I am currently running**

- SpongeForge version: spongeforge-1.12.2-2838-7.1.11-RC4007.jar
- Forge version: forge-1.12.2-14.23.5.2847-universal.jar
- Java version: 1.8.0_212 OpenJ9
- Operating System: Alpine container.

- Plugins/Mods: The 1.12.2 Pack + 
Plugins (30): Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge, FastAsyncWorldEdit, AmberShop, Aura Update Checker, Boop, Buycraft, CatClearLag, EpicBanItem, GriefDefender, HShopPlots, Holograms, ItemChat, LoginMessages, LuckPerms, MMCReboot, Nucleus, PixelAutoMessages, Plan, PlayerDataReset-Plugin, Prism, SkinChanger, SoundControl, SpongePrometheusExporter, Total Economy, VpnBeGone, WorldEdit, spark

**Issue Description**
Started busy server in whitelist mode, after 10~ or so connection attempts the main thread locked up.

```
> [14:00:12] [Server thread/INFO] [FML]: [Server thread] Server side modded connection established
> [14:00:12] [Server thread/INFO] [minecraft/PlayerList]: Disconnecting com.mojang.authlib.GameProfile@a63269c1[id=3cbbe093-c9d1-4c83-89ca-102f684d6541,name=ShlapNutz,properties={textures=[Property{name=textures, value=eyJ0aW1lc3RhbXAiOjE1ODUwNTgzODcxMTEsInByb2ZpbGVJZCI6IjNjYmJlMDkzYzlkMTRjODM4OWNhMTAyZjY4NGQ2NTQxIiwicHJvZmlsZU5hbWUiOiJTaGxhcE51dHoiLCJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYTRlMzJjZjZiYWFmMjlkMmI5NzIyZjg1Yzk5MGQ2OWZjZTFiZGVmNjk2MDZlMTVjMDRjMmVlN2YzYjQ3ZTZmYSJ9fX0=, signature=F+r5ugbLS0g1KVSvBe6yhbWcbf5NIikFcg1hby0fWWpNw/RF2lxCX/9DvRmeDpU3CjJc6AoIpd+UDOvA4lGLvt1+YllzivvMHHTfitTX+MyAtGCfORqLIPNsHRYPBUaJnlFTEseLiZIp9quJQ3A6cpGGBC7viI0wDS0k32CBgDBEnHLislLC/qvYpsibdDSfX4atf+SMRn2PYMHu71m8i9LX5rkCKtyuzI9brXnb9tt8anjTh0JitQpYFYlpHFzW5RXyKghGkzFGGNmEjNHQ3KAKIqu41dzQyM/0toyeTCxN0K78aP/KoEwc90L/zGa1jtbztQU0Hs/1/uZxx5tzrOj7DdDOCPS2t3XnXazCFRf9JpaCPEvcNEhL+c0o11kxF7rcIPHlEZYUxaPxz58L79ru1vFvsFsYiNwN0Mhfkvsh/TB5+ufv49esgL8Gr58tE+fuHRuOcVDxKXDqkv7b/dTQd/HmSLz7lxhm3PqyA/IUy6DcPBBHVT/kNnW64Orr7LtC2bLclimKoo4hlUUikMp++zfBQ+6AjxdxNfqnJPQZSyTRY0KQL+nyv2v9rua7HazCD3GayxfgseJqwNHSOwuW4RDaZAFK/4WZBhTpxR8iWGygWGkJCaaXoJrE8xqH4J48vxL6xQ9slkkUcwMijKOWs5DTk6LqQmVKMNU5T1Q=}]},legacy=false] (/:65271)
> [14:00:42] [Sampler watchdog thread/WARN] [sampler]: The watchdog detected that execution has been stalled for 30 s (#1).
> [14:00:42] [Sampler watchdog thread/INFO] [sampler]: Stall report created at /home/container/stall-reports/stall-2020-03-24_14.00.42.txt.
> [14:01:12] [Sampler watchdog thread/WARN] [sampler]: The watchdog detected that execution has been stalled for 60 s (#1).
> [14:01:12] [Sampler watchdog thread/INFO] [sampler]: Stall report created at /home/container/stall-reports/stall-2020-03-24_14.01.12-1.txt.
> [14:01:54] [pool-3-thread-18/INFO] [buycraft]: Fetching all due players...
> [14:01:54] [pool-3-thread-18/INFO] [buycraft]: Fetched due players (0 found).
> [14:02:12] [Sampler watchdog thread/WARN] [sampler]: The watchdog detected that execution has been stalled for 120 s (#1).
> [14:02:13] [Sampler watchdog thread/INFO] [sampler]: Stall report created at /home/container/stall-reports/stall-2020-03-24_14.02.12-2.txt.
> [14:03:54] [pool-3-thread-18/INFO] [buycraft]: Fetching all due players...
> [14:03:55] [pool-3-thread-18/INFO] [buycraft]: Fetched due players (0 found).
> [14:03:59] [pool-3-thread-15/INFO] [minecraft/DedicatedServer]: [CatClearLag] Ground items will be cleared in 60 seconds.
> [14:04:13] [Sampler watchdog thread/WARN] [sampler]: The watchdog detected that execution has been stalled for 240 s (#1).
> [14:04:13] [Sampler watchdog thread/INFO] [sampler]: Stall report created at /home/container/stall-reports/stall-2020-03-24_14.04.13-3.txt.
> [14:04:29] [pool-3-thread-15/INFO] [minecraft/DedicatedServer]: [CatClearLag] Ground items will be cleared in 30 seconds.
> [14:05:55] [pool-3-thread-36/INFO] [buycraft]: Fetching all due players...
> [14:05:55] [pool-3-thread-36/INFO] [buycraft]: Fetched due players (0 found).
> [14:07:55] [pool-3-thread-33/INFO] [buycraft]: Fetching all due players...
> [14:07:56] [pool-3-thread-33/INFO] [buycraft]: Fetched due players (0 found).
> [14:08:13] [Sampler watchdog thread/WARN] [sampler]: The watchdog detected that execution has been stalled for 480 s (#1).
> [14:08:13] [Sampler watchdog thread/INFO] [sampler]: Stall report created at /home/container/stall-reports/stall-2020-03-24_14.08.13-4.txt.
```

Stall report: https://gist.github.com/clienthax/4f2927eeffb1cd7fef7af29d51b9982e

Seems this is a rare lockup, but seems to be in sponges calls.

**Comment by JBYoshi** [comment]
Looks like a deadlock of some form. The main thread is holding a lock on `NetworkSystem.networkManagers` and waiting on a Netty close operation, while the Netty processing threads are blocked trying to add connections to `networkManagers`.

**Comment by Beshelmek** [comment]
Same issue

**Comment by noremac09** [comment]
Same issue. I also have this problem when many players can't connect to a full server.

    Minecraft: 1.12.2
    SpongeAPI: 7.2-fe189dcf
    Sponge: 1.12.2-7.2.0
    SpongeForge: 1.12.2-2838-7.2.0
    Minecraft Forge: 14.23.5.2838
    JVM: 1.8.0_221/64-bit (Oracle Corporation)
    OS: Linux (4.19.0-8-amd64/amd64)

Stall reports: https://pastebin.com/ggDWLY5A and https://pastebin.com/ehC8RiVf

**Comment by LXGaming** [comment]
Should be fixed in SpongeForge-1.12.2-2838-7.2.1-RC4017
